apps/ui/src/composables/useActions.ts

```diff
@@ -126,7 +133,13 @@ export function useActions() {
   }
 
   async function getAliasSigner() {
-    const network = getNetwork(metadataNetwork);
+    const network = getNetwork(
+      STARKNET_CONNECTORS.includes(web3.value.type as Connector)
+        ? (starknetNetworks.find(id =>
+            enabledNetworks.includes(id)
+          ) as NetworkID)
+        : metadataNetwork
+    );
```

Comment:
With ArgentX [validating `domain.chainId`](https://github.com/snapshot-labs/sx-monorepo/pull/383#issuecomment-2250216427) I think we shouldn't leave the fact if you can or cannot use given wallet to just the order of items in `starknetNetworks` - if someone wants to use different wallet then they need to:
1. know that this array controls it
2. modify order themselves

I think it's confusing.

Adding yet another .env to control it will be a bit much as well, but I liked previous approach of having map from `metadataNetwork` to `starknetNetwork`. This way if I'm dealing with mainnet networks Starknet mainnet will work, if I'm dealing with testnet networks Starknet sepolia will work, quite easily controlled by single variable.

Can we bring it back?

Reply:
Back

Reply:
When asking the wallet to sign the message, the wallet can still be on the wrong network, will be fixed via another PR https://github.com/snapshot-labs/sx-monorepo/issues/546